src/crates.rs

```diff
@@ -137,116 +149,97 @@ pub fn sync_crates_files(
                 }
             }
 
+            // Get the data for this crate file
             let oid = df.id();
             if oid.is_zero() {
-                // The crate was removed, continue to next crate
+                // The crate was removed, continue to next crate.
+                // Note that this does not include yanked crates.
+                removed_crates.push(p.to_path_buf());
                 return true;
             }
             let blob = repo.find_blob(oid).unwrap();
             let data = blob.content();
-            count += Cursor::new(data).lines().count();
+
+            // Download one crate for each of the versions in the crate file
+            for line in Cursor::new(data).lines() {
+                let line = line.unwrap();
+                let c: CrateEntry = serde_json::from_str(&line).unwrap();
+
+                changed_crates.push(c);
+            }
+
             true
         },
         None,
         None,
         None,
-    )?;
+    )
+    .unwrap();
 
-    let (pb_thread, sender) = progress_bar(Some(count), prefix);
+    pb.finish_and_clear();
+    let pb = ProgressBar::new(changed_crates.len() as u64)
+        .with_style(
+            ProgressStyle::default_bar()
+                .template("{prefix} {wide_bar} {pos}/{len} [{elapsed_precise} / {duration_precise}]")
+                .progress_chars("█▉▊▋▌▍▎▏  "),
+        )
+        .with_prefix(prefix);
+    pb.enable_steady_tick(10);
+    pb.set_draw_rate(10);
 
-    let mut removed_crates = vec![];
+    let tasks = futures::stream::iter(changed_crates.into_iter())
+        .map(|c| {
+            // Duplicate variables used in the async closure.
+            let path = path.to_owned();
+            let mirror_retries = mirror.retries;
+            let crates_source = crates_source.map(|s| s.to_string());
+            let user_agent = user_agent.to_owned();
+            let pb = pb.clone();
+            let c = c.to_owned();
 
-    // Download crates multithreaded
-    Pool::new(crates.download_threads as u32).scoped(|scoped| {
-        diff.foreach(
-            &mut |delta, _| {
-                let df = delta.new_file();
-                let p = df.path().unwrap();
-                if p == Path::new("config.json") {
-                    return true;
-                }
+            tokio::spawn(async move {
+                pb.inc(1);
 
-                // DEV: if dev_reduced_crates is enabled, only download crates that start with z
```

Comment:
It's actually still there - I just refactored this code so that instead of two passes in `diff.foreach`, it does one pass and collects a list of crates to change/update.
When building that list, it will still check for this feature and act accordingly.

Reply:
Ohhhh! I see the other instance still in the code. Perfect!